kernel/src/hil/gpio.rs

```diff
@@ -137,6 +153,27 @@ pub trait Input {
     /// pin, return the output; for an input pin, return the input;
     /// for disabled or function pins the value is undefined.
     fn read(&self) -> bool;
+
+    /// Get the current state of a button GPIO pin, for a given button mode.
+    fn read_button(&self, mode: ButtonMode) -> ButtonState {
```

Comment:
Something like `read_activation` could be the generic version of this that works for LEDs and Buttons, I think?